pkg/agent/agent.go

```diff
@@ -118,8 +119,12 @@ func (a *Agent) Run() error {
 	for {
 		select {
 		case err = <-a.config.ErrorCh:
+			e := a.Shutdown()
+			if e != nil {
+				return e
```

Comment:
Feels like we should log this instead so we still get to know the original error